### PR TITLE
Replace `CslStringList` `From` implementation with `TryFrom`

### PR DESCRIPTION
**Pull request by lnicola**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

This isn't _that_ ugly, and I think it's better than panicking in a `From` implementation.

### CODE REVIEW

File: src/cpl.rs [hunk 1]
```diff
@@ -166,44 +166,25 @@ impl Debug for CslStringList {
...
-///
-/// ```rust, no_run
-/// use gdal::cpl::CslStringList;
-/// fn count_opts<O: Into<CslStringList>>(opts: O) -> usize {
-///     opts.into().len()
-/// }
-///
-/// assert_eq!(count_opts(()), 0);
-/// ```
-impl From<()> for CslStringList {
```
Comment (metasim) [inline src/cpl.rs:182]: 😢

File: src/vector/geometry.rs [hunk 2]
```diff
@@ -692,7 +688,7 @@ mod tests {
     /// Simple clone case.
     pub fn test_make_valid_clone() {
         let src = Geometry::from_wkt("POINT (0 0)").unwrap();
-        let dst = src.make_valid(());
+        let dst = src.make_valid(&CslStringList::new());
```
Comment (metasim) [inline src/vector/geometry.rs:691]: 😢

**Review by metasim — APPROVED** [review]
Looks like 1.67 came with new clippy lints...

**Comment by metasim** [comment]
bors d+

**Comment by lnicola** [comment]
There's another PR for the lints.

File: src/cpl.rs (hunk 1 above)
Comment (lnicola) [reply src/cpl.rs:180]: This isn't so bad, is it? I mean, it's `from(&[])` vs. `from(())`..

Comment (metasim) [reply src/cpl.rs:180]: Nah, I'm just being silly.  I was bothered by the verbosity of `src.make_valid(&CslStringList::new())` (vs. `src.make_valid(())`), but fell into being overly clever and not really common/idiomatic. I understand why you got rid of it.

**Comment by lnicola** [comment]
bors d=metasim

**Comment by lnicola** [comment]
bors r=metasim

**Pull request merged**